internal/service/dynamodb/global_secondary_index_test.go

```diff
@@ -0,0 +1,2248 @@
+					"global_secondary_index",
+				},
+				ImportStateCheck: acctest.ComposeAggregateImportStateCheckFunc(
+					acctest.ImportCheckResourceAttr(names.AttrName, rNameTable),
+					acctest.ImportCheckResourceAttr("billing_mode", "PROVISIONED"),
+					acctest.ImportCheckResourceAttr("deletion_protection_enabled", acctest.CtFalse),
+					acctest.ImportCheckResourceAttr("attribute.#", "2"),
+					acctest.ImportCheckResourceAttr("attribute.0.%", "2"),
+					acctest.ImportCheckResourceAttr("attribute.0.type", "S"),
+					acctest.ImportCheckResourceAttrSet("attribute.0.name", true),
+					acctest.ImportCheckResourceAttr("attribute.1.%", "2"),
+					acctest.ImportCheckResourceAttr("attribute.1.type", "S"),
+					acctest.ImportCheckResourceAttrSet("attribute.1.name", true),
+					acctest.ImportCheckResourceAttr("global_secondary_index.#", "1"),
+					acctest.ImportCheckResourceAttr("hash_key", rNameTable),
+					acctest.ImportCheckResourceAttr("local_secondary_index.#", "0"),
+					acctest.ImportCheckResourceAttr("point_in_time_recovery.#", "1"),
+					acctest.ImportCheckResourceAttr("point_in_time_recovery.0.enabled", acctest.CtFalse),
+					acctest.ImportCheckResourceAttrSet("range_key", false),
+					acctest.ImportCheckResourceAttr("replica.#", "0"),
+					acctest.ImportCheckResourceAttrSet("restore_date_time", false),
+					acctest.ImportCheckResourceAttrSet("restore_source_name", false),
+					acctest.ImportCheckResourceAttrSet("restore_to_latest_time", false),
+					acctest.ImportCheckResourceAttr("server_side_encryption.#", "0"),
+					acctest.ImportCheckResourceAttr(names.AttrStreamARN, ""),
+					acctest.ImportCheckResourceAttr("stream_enabled", acctest.CtFalse),
+					acctest.ImportCheckResourceAttr("stream_label", ""),
+					acctest.ImportCheckResourceAttr("stream_view_type", ""),
+					acctest.ImportCheckResourceAttr("table_class", "STANDARD"),
+					acctest.ImportCheckResourceAttr(acctest.CtTagsPercent, "0"),
+					acctest.ImportCheckResourceAttr(acctest.CtTagsAllPercent, "0"),
+					acctest.ImportCheckResourceAttr("read_capacity", "1"),
+					acctest.ImportCheckResourceAttr("write_capacity", "1"),
+				),
+			},
+			{
+				ResourceName:      resourceNameGSI,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"read_capacity",
+					"write_capacity",
+				},
+				ImportStateCheck: acctest.ComposeAggregateImportStateCheckFunc(
+					acctest.ImportCheckResourceAttr(names.AttrName, rNameGSI),
+					acctest.ImportCheckResourceAttr("table", rNameTable),
+					acctest.ImportCheckResourceAttr("hash_key", rNameTable),
+					acctest.ImportCheckResourceAttr("hash_key_type", "S"),
+					acctest.ImportCheckResourceAttr("range_key", rNameGSI),
+					acctest.ImportCheckResourceAttr("range_key_type", "S"),
+					acctest.ImportCheckResourceAttr("projection_type", "ALL"),
+					acctest.ImportCheckResourceAttr("read_capacity", "1"),
+					acctest.ImportCheckResourceAttr("write_capacity", "1"),
+				),
```

Comment:
`ImportStateCheck` is usually not needed, as `ImportStateVerify: true` will validate against the values from Step 1. If the results are expected to be different, add a comment explaining the expected change